crates/router/src/core/payments/helpers.rs

```diff
@@ -2487,3 +2487,94 @@ mod test {
         );
     }
 }
+
+pub async fn get_additional_payment_data(
+    pm_data: &api_models::payments::PaymentMethodData,
+    db: &dyn StorageInterface,
+) -> RouterResult<api_models::payments::AdditionalPaymentData> {
+    match pm_data {
+        api_models::payments::PaymentMethodData::Card(card_data) => {
+            if card_data.card_issuer.is_some()
+                && card_data.card_network.is_some()
+                && card_data.card_type.is_some()
+                && card_data.card_issuing_country.is_some()
+                && card_data.bank_code.is_some()
+            {
+                Ok(api_models::payments::AdditionalPaymentData::Card {
+                    card_issuer: card_data.card_issuer.to_owned(),
+                    card_network: card_data.card_network.clone(),
+                    card_type: card_data.card_type.to_owned(),
+                    card_issuing_country: card_data.card_issuing_country.to_owned(),
+                    bank_code: card_data.bank_code.to_owned(),
+                })
+            } else {
+                let card_number = card_data.clone().card_number;
+                let card_info = db
+                    .get_card_info(&card_number.get_card_isin())
+                    .await
+                    .attach_printable("Failed to retrieve card information")
+                    .ok()
+                    .flatten();
+                Ok(api_models::payments::AdditionalPaymentData::Card {
+                    card_issuer: card_info.as_ref().and_then(|card| {
+                        card.card_issuer.as_ref().map(|issuer| issuer.to_string())
+                    }),
+                    card_network: card_info.as_ref().and_then(|card| {
+                        card.card_network
+                            .clone()
+                            .map(|network| network.foreign_into())
+                    }),
+                    card_type: card_info
+                        .as_ref()
+                        .and_then(|card| card.card_type.as_ref().map(|issuer| issuer.to_string())),
+                    bank_code: card_info.as_ref().and_then(|card| {
+                        card.bank_code
+                            .as_ref()
+                            .map(|bank_code: &String| bank_code.to_string())
+                    }),
+                    card_issuing_country: card_info.as_ref().and_then(|card| {
+                        card.card_issuing_country
+                            .as_ref()
+                            .map(|card_issuing_country| card_issuing_country.to_string())
+                    }),
+                })
```

Comment:
Can this be simplified by using `and_then` only once?

Reply:
done!